frontend/src/components/AnimatedMapDisplay.jsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useRef } from "react";
+import * as THREE from 'three';
+import { OrbitControls } from 'three/examples/jsm/controls/OrbitControls.js';
+import { GLTFLoader } from 'three/addons/loaders/GLTFLoader.js';
+import { getSteps } from "../utils";
+
+const AnimatedMapDisplay = ({gameMap, tileConfig, path, enemies}) => {
+    const mapRef = useRef();
+    const center = new THREE.Vector3(gameMap.width/2,0,gameMap.width/2);
+    const {heightMap} = gameMap;
+    tileConfig ||= {
+        width: 1,
+        height: 0.25,
+    };
+    const offset = {
+        x: tileConfig.width/2 - center.x,
+        y: tileConfig.height/2,
+        z: tileConfig.width/2 - center.z,
+    };
+    const animationFunctions = [];
+
+    const loadModel = (loader, scene, model, x, y, z, scale = 1, rotation = 0, curryAnimate) => {
+        loader.load(model, (m) => {
+            m.scene.scale.set(scale, scale, scale);
+            m.scene.position.set(
+                x + offset.x,
+                tileConfig.height*y + offset.y,
+                z + offset.z,
+            );
+            m.scene.rotation.y = rotation;
+
+            // Does this make any sense at all?
+            if (curryAnimate) animationFunctions.push(curryAnimate(m.scene));
```

Comment:
Ideally we would have a way to remove the callbacks as well. That way, we could deallocate the models after they reach their destination instead of just making them invisible.